### PR TITLE
Refactor handlers to move common code to libcontainer handler

### PR DESCRIPTION
**Pull request by dashpole**
This should make implementing https://github.com/google/cadvisor/pull/1872 a bit cleaner
cc @nielsole

### CODE REVIEW

File: container/libcontainer/helpers.go [hunk 1]
```diff
@@ -398,8 +414,8 @@ func scanUdpStats(r io.Reader) (info.UdpStat, error) {
 	return stats, nil
 }
 
-func GetProcesses(cgroupManager cgroups.Manager) ([]int, error) {
-	pids, err := cgroupManager.GetPids()
+func (l *Handler) GetProcesses() ([]int, error) {
```
Comment (nielsole) [inline container/libcontainer/helpers.go]: Maybe move all methods of the Handler struct together? Right now they are so scattered throughout the file. Maybe even move them out of `helpers.go`?

Comment (dashpole) [reply container/libcontainer/helpers.go]: done

**Comment by dashpole** [comment]
@nielsole does this look good to you?

**Comment by nielsole** [comment]
I will take a closer look at it tomorrow. On first glance it lgtm

**Comment by nielsole** [comment]
:+1:  lgtm

File: container/containerd/handler.go [hunk 2]
```diff
@@ -36,31 +34,21 @@ import (
...
-	pid int
+	reference info.ContainerReference
+	envs      map[string]string
+	labels    map[string]string
 	// Image name used for this container.
 	image string
-	// The host root FS to read
-	rootFs string
 	// Filesystem handler.
 	ignoreMetrics container.MetricSet
```
Comment (nielsole) [inline container/containerd/handler.go]: We missed removing the `ignoreMetrics` from all handlers

Comment (nielsole) [inline container/containerd/handler.go]: We missed removing the `ignoreMetrics` from all handlers

Comment (nielsole) [reply container/containerd/handler.go]: wrong line actually, aimed for 48-49

Comment (dashpole) [reply container/containerd/handler.go]: you are welcome to do this in your PR.  IIRC ignoreMetrics was still used by the other handlers, but I may be mistaken...

**Pull request merged**